### PR TITLE
Fix: change name for david 0.9.2+

### PR DESCRIPTION
**Pull request by Jamaika1**

### CODE REVIEW

**Comment by joedrago** [comment]
I believe @wantehchang was considering a PR for this name change, but is there a way you can detect when to use the new API, and when to use the old API? The change is definitely _after_ v0.9.2, so we need some sort of ifdef/test to support old and new versions of dav1d.

This might mean that we need to wait for `v0.9.3` and switch on that version, if there is no other signal.

File: src/codec_dav1d.c [hunk 1]
```diff
@@ -57,8 +57,8 @@ static avifBool dav1dCodecGetNextImage(struct avifCodec * codec,
 {
     if (codec->internal->dav1dContext == NULL) {
         // Give all available threads to decode a single frame as fast as possible
-        codec->internal->dav1dSettings.n_frame_threads = 1;
-        codec->internal->dav1dSettings.n_tile_threads = AVIF_CLAMP(decoder->maxThreads, 1, DAV1D_MAX_TILE_THREADS);
+        codec->internal->dav1dSettings.n_threads = 1;
+        codec->internal->dav1dSettings.max_frame_delay = AVIF_CLAMP(decoder->maxThreads, 1, 256);
```
Comment (1480c1) [inline src/codec_dav1d.c:61]: Perhaps it's better to use `DAV1D_MAX_THREADS`  instead of a bare number

Comment (1480c1) [inline src/codec_dav1d.c:61]: You might want to use
```c
#if defined(DAV1D_API_VERSION_MAJOR) && DAV1D_API_VERSION_MAJOR >= 6
// new fields and code here
#else
// old code here
#endif
```
in order to keep compile compatibility with older dav1d versions and installs

Comment (Jamaika1) [reply src/codec_dav1d.c:61]: `DAV1D_MAX_THREADS` is delete in david 0.9.2+. I guess that changed the value from 64 to 256.

Comment (1480c1) [reply src/codec_dav1d.c:62]: I see it at https://code.videolan.org/videolan/dav1d/-/merge_requests/1252/diffs#95378459063068eb66ef004414047378fc28a963_49_46 and at https://code.videolan.org/videolan/dav1d/-/blob/master/include/dav1d/dav1d.h#L46

Comment (Jamaika1) [reply src/codec_dav1d.c]: OK. I added DAV1D_API_VERSION_MAJOR

Comment (1480c1) [reply src/codec_dav1d.c]: This can be marked as resolved

**Comment by wantehchang** [comment]
Hi Jamaika1,

Thank you for writing the pull request. My coworker Michael Bradshaw wrote a patch for internal use at Google yesterday. I just converted his patch to https://github.com/AOMediaCodec/libavif/pull/751.

File: src/codec_dav1d.c (hunk 1 above)
Comment (1480c1) [reply src/codec_dav1d.c]: Wrong macro

**Pull request closed without merging**